guides/scripts/extract_links_toc.rb

```diff
@@ -0,0 +1,26 @@
+#! /usr/bin/ruby
+
+require 'nokogiri'
+require 'json'
+
+files = ARGV
+
+toc = {}
+
+files.each do |file|
+  document = Nokogiri::HTML.parse(File.open(file))
+
+  file_toc = Hash[
+    document.css('h2').map do |chapter|
+      [
+        chapter['id'].downcase,
+        chapter.parent.css('h3').map {|subchapter| subchapter['id']} +
+        chapter.parent.css('div[id]').map {|subchapter| subchapter['id']}
+      ]
+    end
+  ]
```

Comment:
I'd use `to_h`

```suggestion
  file_toc = document.css('h2').to_h do |chapter|
    [
      chapter['id'].downcase,
      chapter.parent.css('h3').map {|subchapter| subchapter['id']} +
      chapter.parent.css('div[id]').map {|subchapter| subchapter['id']}
    ]
  end
```

But then also combine the CSS selectors to make it easier to understand what's going on


```suggestion
  file_toc = document.css('h2').to_h do |chapter|
    [
      chapter['id'].downcase,
      (chapter.parent.css('h3') + chapter.parent.css('div[id]')).map {|subchapter| subchapter['id']}
    ]
  end
```